### PR TITLE
create MethodicalWork resource

### PR DESCRIPTION
**Pull request by zdubzdab**

### CODE REVIEW

File: app/controllers/teachers/methodical_works_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+      if @methodical_work.save
+        redirect_to teachers_methodical_works_path, notice: t('.controllers.sucessful')
+      else
+        render "new"
+      end
+  end
+
+  def index
+    @methodical_works = MethodicalWork.all.order("created_at DESC").page(params[:page]).per(User::TEACHER_METHODICAL_WORKS_INDEX_PAGES)
```
Comment (begin29) [inline app/controllers/teachers/methodical_works_controller.rb]: please use format
```ruby
MethodicalWork.all
                          .order("created_at DESC")
                          .page(params[:page]) ...

File: app/models/ability.rb [hunk 2]
```diff
@@ -14,14 +14,18 @@ def initialize(user)
       can :read, User
       can :read, Test
       can :read, TestSetting
+      can [:edit, :update, :destroy], MethodicalWork, :user_id => user.id
+      can [:read, :create], MethodicalWork
```
Comment (begin29) [inline app/models/ability.rb]: why do we need this splitting? Could we say `can :manage, MethodicalWork` or with `user_id` ? Please try and check

File: app/models/user.rb [hunk 3]
```diff
@@ -10,6 +10,7 @@ class User < ActiveRecord::Base
   ADMIN_USERS_INDEX_PAGES = 5
   USERS_INDEX_PAGES = 5
   ADMIN_TIDINGS_INDEX_PAGES = 20
+  TEACHER_METHODICAL_WORKS_INDEX_PAGES = 20
```
Comment (begin29) [inline app/models/user.rb]: could we rename it to `METODICAL_WORKS_PER_PAGE` ?

File: app/views/teachers/methodical_works/_form.html.slim [hunk 4]
```diff
@@ -0,0 +1,29 @@
+= form_for [:teachers, @methodical_work], html: { class: 'methodical_work-form',  multipart: true } do |f|
```
Comment (begin29) [inline app/views/teachers/methodical_works/_form.html.slim]: change class name with `dashes` instead undescore

File: app/models/user.rb (hunk 3 above)
Comment (zdubzdab) [inline app/models/user.rb]: as for me TEACHER_METHODICAL_WORKS_INDEX_PAGES (namespace/controller name/action) is more clear or it isn't?

Comment (begin29) [inline app/models/user.rb]: what is the main purpose of that constant? It is responsible for representing how many methodical works per page will be shown. Does it make any sense?

File: app/assets/stylesheets/teachers/methodical_works.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+#methodical-work-textarea {
+  height: 120px;
+}
+
+#table-with-methodical-works {
+  position: relative;
+  left: 5%;
+
+  td {
```
Comment (begin29) [inline app/assets/stylesheets/teachers/methodical_works.scss]: whitespace!

File: app/models/ability.rb [hunk 6]
```diff
@@ -14,14 +14,17 @@ def initialize(user)
...
       can :read, TestSetting
+      can :manage, MethodicalWork, user_id: user.id
     elsif user.has_role? :student
       can :create, Comment
       can :read, Tiding
       can :read, User
+      can :read, MethodicalWork
     else
       can :create, Comment
       can :read, Tiding
```
Comment (begin29) [inline app/models/ability.rb]: is it possible to use this construction:

```ruby
can :read, [Comment, User ....]

File: app/views/teachers/methodical_works/_methodical_works.html.slim [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+      th#col-3 = t '.time'
+      th#col-4 = t 'table.attributes.edit'
+      th#col-5 = t 'table.attributes.destroy'
+
+      - @methodical_works.each do |methodical_work|
+        tr
+          td = methodical_work.title
+          td = methodical_work.text
+          td = date_format(methodical_work.created_at)
+          td = link_to t('buttons.edit'),  edit_teachers_methodical_work_path(methodical_work)
```
Comment (begin29) [inline app/views/teachers/methodical_works/_methodical_works.html.slim]: could you add `edit` icon from [here](http://getbootstrap.com/components/)

File: app/views/teachers/methodical_works/_methodical_works.html.slim [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+          td = link_to t('buttons.destroy'), teachers_methodical_work_path(methodical_work), method: :delete, data: { confirm: t('.destroy_certain') }, class: "b-delete-tiding"
```
Comment (begin29) [inline app/views/teachers/methodical_works/_methodical_works.html.slim]: ^^

File: app/assets/stylesheets/teachers/methodical_works.scss [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    max-width: 0;
```
Comment (begin29) [inline app/assets/stylesheets/teachers/methodical_works.scss]: why max-width: 0 ? Do you want to hide this raw?

**Pull request merged**